docs/dependency_tracking.rst

```diff
@@ -0,0 +1,99 @@
+
+Dependency Tracking
+###################
+
+One of the biggest features of rapids-cmake is that is can track dependencies ( `find_package`, `cpm` )
+allowing for projects to easily generate `<project>-config.cmake` files with correct dependency requirements.
+In a normal CMake project public dependencies need to be recorded in two locations, the original ``CMakeLists.txt`` file and also the generated `<project>-config.cmake`. This dual source of truth increases
+developer burden, and adds a common source of error.
+
+``rapids-cmake`` is designed to remove this dual source of truth by expanding the concept of Modern CMake `Usage Requirements <https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements>`_  to include external packages.
+This is done via the ``BUILD_EXPORT_SET`` ( maps to  `<BUILD_INTERFACE>` ) and ``INSTALL_EXPORT_SET`` ( maps to `<INSTALL_INTERFACE>` ) keywords on commands such as :cmake:command:`rapids_find_package` and :cmake:command:`rapids_cpm_find`.
+Lets go over an example on how these come together to make dependency tracking for projects super easy:
+
+.. code-block:: cmake
+
+  rapids_find_package(Threads REQUIRED
```

Comment:
Would it make more sense to integrate the code with the prose descriptions of the code, since this is a walkthrough?

Reply:
Do you think repeating the relevant code blocks again for each section would work? I went with a single code block so people could see the order of the different components, and if needed would be easy to copy.

Maybe the full example it could go at the end?

Reply:
I have updated the docs to repeat the relevant code blocks for each subsection

Reply:
I like it.